File: 2018_prelim/tasks/island/solutions/ranald/st4_ranald2.py
N, M = map(int, raw_input().split())
cnt = [0]*(N+M+1)
for i in range(N+M-1):
	u, v = map(int, raw_input().split())
	cnt[u] = cnt[u] + 1
	cnt[v] = cnt[v] + 1
	
fac = [0]*(N+M+1)
for i in range(1, N+M+1):
	for k in range(2, cnt[i]):
		print ("%d %d" % (k, 1))

File: 2018_prelim/tasks/island/solutions/ranald/st4_ranald2.cpp
#include <bits/stdc++.h>
using namespace std;
const int MAXN = 200004;
int N, M, cnt[MAXN], fac[MAXN];
int main() {
	scanf("%d%d", &N, &M);
	for (int i = 0, u, v; i < N+M-1; ++i) {
		scanf("%d%d", &u, &v);
		cnt[u-1]++;
		cnt[v-1]++;
	}
	for (int i = 0; i < N+M; ++i) {
		for (int k = 2; k < cnt[i]; ++k) {
			printf("%d 1\n", k);
		}
	}

}
